Created documentation for `MiddleTextTruncation` (#2329)
iTwin/iTwinUI: apps/website/src/content/docs/middletexttruncation.mdx

<LiveExample src='MiddleTextTruncation.main.jsx' truncated={false}>
  <AllExamples.MiddleTextTruncationMainExample client:load />
</LiveExample>

For more examples, see [truncation on Select](/docs/select#truncation).

### Number of end characters

By default, six characters are left visible at the end of the truncated text. This ensures that important suffixes, such as file extensions or identifiers, remain visible. This number can be modified through the `endCharsCount` prop to suit user's specific needs.

<LiveExample src='MiddleTextTruncation.endCharsCount.jsx'>
  <AllExamples.MiddleTextTruncationEndCharsCountExample client:load />
</LiveExample>

### Customization

For further control over the rendering of the content, the `textRenderer` prop allows users to provide a custom rendering function. This function takes `originalText` and `truncatedText` as parameters and returns a `ReactNode`.

<LiveExample src='MiddleTextTruncation.customizeTruncation.jsx'>
  <AllExamples.MiddleTextTruncationCustomizationExample client:load />
</LiveExample>

> ### Accessibility

Reviewer: This doesn't belong under "Usage" so it should be an h2

```suggestion
## Accessibility
```

> This component only truncates the content visually but leaves the full content in the accessibility tree. That way, users using assistive technologies can still have access to the full content which is consistent with the default overflow behavior.

Reviewer: ```suggestion
This component only truncates the content visually but leaves the full content in the accessibility tree. That way, users using assistive technologies can still have access to the full content. This is also consistent with the browser's default `text-overflow` behavior.
```

Author: I have made the change.

---
title: MiddleTextTruncation
description: A utility component used for truncating texts which exceed certain limit.
thumbnail: #TODO
group: utilities
---

<p>{frontmatter.description}</p>

## Usage

In circumstances where a fixed width constraint is required, the `MiddleTextTruncation` component can be helpful in handling long text strings by truncating them in the middle by inserting the ellipsis (`"..."`). This ensures that the beginning and the end of the text remain visible.

<LiveExample src='MiddleTextTruncation.main.jsx' truncated={false}>
  <AllExamples.MiddleTextTruncationMainExample client:load />
</LiveExample>

> For more application examples, see this [documentation on Select](/docs/select#truncation).

Author: I have made the change.

Adjusting this prop allows end users to control how much of the end of the text is shown, which can be particularly useful for maintaining readability and context in various use cases.

<LiveExample src='MiddleTextTruncation.endCharsCount.jsx'>
  <AllExamples.MiddleTextTruncationEndCharsCountExample client:load />
</LiveExample>

### Customization

For further customization, the `textRenderer` prop allows end users to define a custom rendering function for the truncated text. This function receives two parameters:

- `originalText`
- `truncatedText`

The function should return a `ReactNode` that represents the customized text element. This is particularly useful when end users want to apply specific styles or add additional elements to the truncated text, such as highlighting the truncated portion or adding tooltips for better user experience.

<LiveExample src='MiddleTextTruncation.customizeTruncation.jsx'>
  <AllExamples.MiddleTextTruncationCustomizationExample client:load />
</LiveExample>

### Accessibility

> This component is designed for visual truncation and should not impact the accessibility tree, as the truncated text would be confusing, especially for non-sighted users. The original text is kept visually hidden, and the truncated text is removed from the accessibility tree. This approach ensures consistency with the built-in `text-overflow: ellipsis`, allowing the same UX design guidelines to be applied in both cases.

Author: All the wording suggestions above make sense to me, so I have made all changes.
